tests/Core/Tokenizer/ReadonlyTest.inc

```diff
@@ -0,0 +1,53 @@
+<?php
+
+class Foo
+{
+    /* testReadonlyProperty */
+    readonly int $readonlyProperty;
+    /* testPublicReadonlyProperty */
+    public readonly int $publicReadonlyProperty;
+    /* testProtectedReadonlyProperty */
+    protected readonly int $protectedReadonlyProperty;
+    /* testPrivateReadonlyProperty */
+    private readonly int $privateReadonlyProperty;
+    /* testPublicReadonlyPropertyWithReadonlyFirst */
+    readonly public int $publicReadonlyProperty;
+    /* testProtectedReadonlyPropertyWithReadonlyFirst */
+    readonly protected int $protectedReadonlyProperty;
+    /* testPrivateReadonlyPropertyWithReadonlyFirst */
+    readonly private int $privateReadonlyProperty;
+    /* testReadonlyWithCommentsInDeclaration */
+    private /* Comment */ readonly /* Comment */ int /* Comment */ $readonlyPropertyWithCommentsInDeclaration;
+
```

Comment:
Could some tests be added with:
* The "old-school" `var` keyword.
* The `static` keyword in varying order (with the keywords in vaying orders).
* No type declaration.
* And could the types used be varied a bit ? - I'd particularly like to see the `array` type be included in the tests, both as singular type as well as the first in a union type, as that's been problematic before (and a check that that `array` type is still re-tokenized to `T_STRING` when used in combination with `readonly).
    You may (also) want to add those to the existing `ArrayKeywordTest` and possibly add some tests with `readonly` to the `BitwiseOrTest` as well.

Reply:
- `readonly` cannot be used with `var`
- It cannot be used in `static` property
- The type is required for readonly properties

However I have to add more types, eg. `parent`, `self`, `?int`, `int|null`, `array`.

Reply:
> * `readonly` cannot be used with `var`
>     * It cannot be used in `static` property
>     * The type is required for readonly properties

The thing is: whether or not it can _legally_ be used like that, users new to PHP 8.1 _may_ still try. If the tokenization of this is not adjusted to take these "incorrect" uses into account:
* Sniffs will not be able to _flag_ those incorrect uses.
* The `readonly` keyword would be grouped in with the type declaration or seen _as_ the type declaration when using the `File::getMemberProperties()` method.
* etc...

All in all, handling `readonly` consistently by allowing for typical mistakes devs may make, will make life a lot easier on sniff writers.

Keep in mind: sniff analysis is token based, not parser based, so parse errors (including incomplete code during live coding) should not break sniffs.

Reply:
Note: no matter what, the tests should be added anyway to document how the PHPCS tokenizer handles these situations.

Reply:
> Keep in mind: sniff analysis is token based, not parser based, so parse errors (including incomplete code during live coding) should not break sniffs.

Yes, but I think it's better to not convert string to invalid `readonly`. In this way it will break less things.

Reply:
However, it looks to copy the PHP 8.1 behavier it's necessary to support it: https://3v4l.org/oEbgM

Reply:
Done.